configx/provider_watch_test.go

```diff
@@ -69,51 +68,40 @@ func lsof(t *testing.T, file string) string {
 		return ""
 	}
 	var b, be bytes.Buffer
-	c := exec.Command("lsof", "-n")
+	// we are only interested in the file descriptors, so we use the `-F f` option
+	c := exec.Command("lsof", "-n", "-F", "f", file)
```

Comment:
I figured out why the output of `lsof` was so unstable @aeneasr.
This is what I get during testing:
```
$ lsof -n | grep -E 'USER|/tmp/1675786139551899475/config-141427151.yml'
COMMAND      PID    TID TASKCMD     USER   FD      TYPE             DEVICE   SIZE/OFF       NODE NAME
___TestRe 518705                  patrik    3u      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705                  patrik    8r      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518736 ___TestRe patrik    3u      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518736 ___TestRe patrik    8r      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518737 ___TestRe patrik    3u      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518737 ___TestRe patrik    8r      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518738 ___TestRe patrik    3u      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518738 ___TestRe patrik    8r      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518739 ___TestRe patrik    3u      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518739 ___TestRe patrik    8r      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518740 ___TestRe patrik    3u      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518740 ___TestRe patrik    8r      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518741 ___TestRe patrik    3u      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518741 ___TestRe patrik    8r      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518742 ___TestRe patrik    3u      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518742 ___TestRe patrik    8r      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518743 ___TestRe patrik    3u      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518743 ___TestRe patrik    8r      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518744 ___TestRe patrik    3u      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518744 ___TestRe patrik    8r      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518745 ___TestRe patrik    3u      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518745 ___TestRe patrik    8r      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518746 ___TestRe patrik    3u      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518746 ___TestRe patrik    8r      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518747 ___TestRe patrik    3u      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518747 ___TestRe patrik    8r      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518748 ___TestRe patrik    3u      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518748 ___TestRe patrik    8r      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518749 ___TestRe patrik    3u      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518749 ___TestRe patrik    8r      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518750 ___TestRe patrik    3u      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518750 ___TestRe patrik    8r      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518751 ___TestRe patrik    3u      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518751 ___TestRe patrik    8r      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518752 ___TestRe patrik    3u      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518752 ___TestRe patrik    8r      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518753 ___TestRe patrik    3u      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518753 ___TestRe patrik    8r      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518754 ___TestRe patrik    3u      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518754 ___TestRe patrik    8r      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518755 ___TestRe patrik    3u      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518755 ___TestRe patrik    8r      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518756 ___TestRe patrik    3u      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518756 ___TestRe patrik    8r      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518757 ___TestRe patrik    3u      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518757 ___TestRe patrik    8r      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml

# close one file descriptor

$ lsof -n | grep -E 'USER|/tmp/1675786139551899475/config-141427151.yml'
COMMAND      PID    TID TASKCMD     USER   FD      TYPE             DEVICE   SIZE/OFF       NODE NAME
___TestRe 518705                  patrik    3u      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518736 ___TestRe patrik    3u      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518737 ___TestRe patrik    3u      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518738 ___TestRe patrik    3u      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518739 ___TestRe patrik    3u      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518740 ___TestRe patrik    3u      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518741 ___TestRe patrik    3u      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518742 ___TestRe patrik    3u      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518743 ___TestRe patrik    3u      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518744 ___TestRe patrik    3u      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518745 ___TestRe patrik    3u      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518746 ___TestRe patrik    3u      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518747 ___TestRe patrik    3u      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518748 ___TestRe patrik    3u      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518749 ___TestRe patrik    3u      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518750 ___TestRe patrik    3u      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518751 ___TestRe patrik    3u      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518752 ___TestRe patrik    3u      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518753 ___TestRe patrik    3u      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518754 ___TestRe patrik    3u      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518755 ___TestRe patrik    3u      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518756 ___TestRe patrik    3u      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
___TestRe 518705 518757 ___TestRe patrik    3u      REG               0,35         26     113048 /tmp/1675786139551899475/config-141427151.yml
```

As you can see, the PID is the same for all output lines, but there is one entry per system-level thread. Go uses threads depending on the number of CPUs below it's goroutines, so obviously this output is completely different on every machine. What we are interested instead is the number of distinct file descriptors, which we reliably get using my change. The previous approach was just measuring random numbers that were completely unrelated to the actual test.